static/app/components/modals/privateGamingSdkAccessModal.tsx

```diff
@@ -0,0 +1,184 @@
+import {Fragment, useCallback, useState} from 'react';
+import {captureFeedback} from '@sentry/react';
+
+import {addSuccessMessage} from 'sentry/actionCreators/indicator';
+import {type ModalRenderProps} from 'sentry/actionCreators/modal';
+import {Button} from 'sentry/components/core/button';
+import {ButtonBar} from 'sentry/components/core/button/buttonBar';
+import SelectField from 'sentry/components/forms/fields/selectField';
+import TextField from 'sentry/components/forms/fields/textField';
+import {t, tct} from 'sentry/locale';
+import type {Organization} from 'sentry/types/organization';
+import {useUser} from 'sentry/utils/useUser';
+
+const PRIVATE_GAMING_SDK_OPTIONS = [
+  {value: 'playstation', label: 'PlayStation'},
+  {value: 'xbox', label: 'Xbox'},
+  {value: 'nintendo-switch', label: 'Nintendo Switch'},
+] as const;
+
+type GamingPlatform = (typeof PRIVATE_GAMING_SDK_OPTIONS)[number]['value'];
+
+export interface PrivateGamingSdkAccessModalProps {
+  organization: Organization;
+  projectSlug: string;
+  sdkName: string;
+  gamingPlatform?: GamingPlatform;
+  onSubmit?: () => void;
+}
+
+export function PrivateGamingSdkAccessModal({
+  Header,
+  Body,
+  Footer,
+  closeModal,
+  organization,
+  projectSlug,
+  sdkName,
+  gamingPlatform,
+  onSubmit,
+}: PrivateGamingSdkAccessModalProps & ModalRenderProps) {
+  const user = useUser();
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [githubProfile, setGithubProfile] = useState('');
+  const [gamingPlatforms, setGamingPlatforms] = useState<string[]>(
+    gamingPlatform ? [gamingPlatform] : []
+  );
+
+  const isFormValid = !!githubProfile.trim() && gamingPlatforms.length > 0;
+
+  const handleSubmit = useCallback(() => {
```

Comment:
can we please _not_ wrap every function we create in `useCallback` 🙏 . There is absolutely no need here, because:

- it’s only passed to the click handler of a `Button`, which doesn’t care about referential stability.
- it is dependent on `onSubmit`, which is a prop, which is _always_ an inline function, so the `useCallback` has to run every time anyways.

All it does in this instance is add more code and complexity and mental overhead.

Reply:
hey thanks! it is not always super clear when we should use useCallback, and I noticed we've been applying it by default whenever we define a function in our codebase. This one for example was copied and pasted from [here](https://github.com/getsentry/sentry/blob/a0f080adc593e349373b3da55adcf7ccd8bffbe5/static/app/views/settings/project/tempest/PlayStationSdkAccessModal.tsx#L51).

I was wondering...do we have a linter or any tooling that could help us identify when useCallback is actually necessary? That would be super helpful!